pkg/kubelet/server/remotecommand/exec.go

```diff
@@ -51,8 +85,19 @@ func ServeExec(w http.ResponseWriter, req *http.Request, executor Executor, podN
 
 	err := executor.ExecInContainer(podName, uid, container, cmd, ctx.stdinStream, ctx.stdoutStream, ctx.stderrStream, ctx.tty, ctx.resizeChan)
 	if err != nil {
-		msg := fmt.Sprintf("error executing command in container: %v", err)
-		runtime.HandleError(errors.New(msg))
-		fmt.Fprint(ctx.errorStream, msg)
+		m := ExitError{}
```

Comment:
We should not be creating a new error type, when we already have one (`APIStatus`).  That's what we should be returning here, and we need to obey versioning rules w.r.t. the error (define in the protocol what sort of error we return).

I'm also concerned that this change is not backwards compatible, in that old clients will get a new, unexpected value on the error stream.